tket/tests/test_RoutingMethod.cpp

```diff
@@ -172,17 +190,20 @@ SCENARIO("Test RoutingMethodCircuit::route_method") {
     Architecture arc(
         {{Node("t", 1), Node("t", 0)}, {Node("t", 2), Node("t", 1)}});
     ArchitecturePtr shared_arc = std::make_shared<Architecture>(arc);
-    unit_map_t output = rmc.routing_method(mf, shared_arc), empty;
-    REQUIRE(output == empty);
+    std::pair<bool, unit_map_t> output = rmc.routing_method(mf, shared_arc);
+    unit_map_t empty;
+    REQUIRE(output.first);
+    REQUIRE(output.second == empty);
 
-    Circuit comp1(2);
+    Circuit comp1(3);
```

Comment:
The "dummy" `test_routing_metohd_mf_swap_perm` has a condition`if (c.n_qubits() > 2 && a->n_nodes() > 2) {` determining whether it makes any changes. In the previous test this case was violated, however the "dummy" method could work regardless and so this just tested the final routing_method. This was fine because these `RoutingMethodCircuit` options are not for general use, but purely for testing.

I decided in this case that changing the number of qubits from 2 to 3 preserved the old functionality being tested. I could have, including whether the routing_method is used or not, and so it was a fine change to make. The alternative would be to change the condition, however the condition itself is not whats being tested, but whether the code flow suitably rejects a `RoutingMethodCircuit` result if the returned boolean is false.